### PR TITLE
Allow unprivileged mode when running as splunk user

### PR DESCRIPTION
**Pull request by mikedickey**
This PR enables running in unprivileged mode when run as the splunk user account. The splunk user is now part of the ansible group and able to run the ansible playbooks itself at startup, but unlike ansible, it has no sudo capabilities at startup time.

Note that you currently must explicitly set SPLUNK_HOME_OWNERSHIP_ENFORCEMENT to false when running as the splunk user, otherwise it will fail due to lack of permissions.

Note that there are limitations, beyond just being unable to "correct" permission for volume mounts. Any features requiring elevated permissions, such as install JDK or other packages, will not work when running as the splunk user.

The defaults are left unchanged. By default, it will still run as the ansible user and the behavior should be the same as before.

### CODE REVIEW

**Comment by mikedickey** [comment]
Also note I made some indenting fixes in entrypoint.sh so that it's (I think) all using tabs now

File: splunk/common-files/entrypoint.sh [hunk 1]
```diff
@@ -89,58 +95,58 @@ configure_multisite() {
...
 	if [[ "$STEPDOWN_ANSIBLE_USER" == "true" ]]; then
-    	bash -c "sudo deluser -q ansible sudo"
+		bash -c "sudo deluser -q ansible sudo"
 	fi
 }
 
 help() {
 	cat << EOF
-  ____        _             _      __
+	____        _             _      __
```
Comment (bb03) [inline splunk/common-files/entrypoint.sh]: I think the tab to spaces messed with our ascii art here.  If we're going all in on tabs, can you fix the spacing here.

**Comment by nwang92** [comment]
Overall, this follows what @lephino and I came up with last Friday so I don't have any real issues with it. I can add some tests for any upgrade functionality so we have that in our pipeline in the future.

File: splunk/common-files/entrypoint.sh [hunk 2]
```diff
@@ -35,6 +35,9 @@ trap teardown SIGINT SIGTERM
 
 prep_ansible() {
 	cd ${SPLUNK_ANSIBLE_HOME}
+	if [ `whoami` == "${SPLUNK_USER}" ]; then
+		sed -i -e "s,^become\\s*=.*,become = false," ansible.cfg
```
Comment (nwang92) [inline splunk/common-files/entrypoint.sh:39]: One small thing here that I don't quite like is that our `ansible.cfg` as shown in the repo is not actually representative of the ansible.cfg during execution under a different user. But I suppose this is a necessary evil at the moment so as long as it's documented, it should be alright.

Comment (mikedickey) [reply splunk/common-files/entrypoint.sh:39]: Yeah, I was hoping that `become_user` when you are that user is a noop, but unfortuantely that is not the case. Without explicitly disabling it here, ansible bombs out at startup with complaints that it cannot use sudo without a password.

**Review by nwang92 — APPROVED** [review]

**Review by lephino — APPROVED** [review]

**Review by arctan5x — APPROVED** [review]

**Review by bb03 — APPROVED** [review]

**Comment by sayeedc** [comment]
I’ve encountered issues running Splunk SE and UF on Openshift in unprivileged mode. When will this PR be put into an image release on either the Redhat container catalog or Docker Hub?

**Comment by mikedickey** [comment]
> I’ve encountered issues running Splunk SE and UF on Openshift in unprivileged mode. When will this PR be put into an image release on either the Redhat container catalog or Docker Hub?

Hi Sayeed, this should be included in the `edge` containers (which are built nightly), and will be included with the next release (7.3.2?). For OpenShift, I'd recommend trying specifically `splunk/splunk:edge-redhat`.

**Comment by sayeedc** [comment]
> > I’ve encountered issues running Splunk SE and UF on Openshift in unprivileged mode. When will this PR be put into an image release on either the Redhat container catalog or Docker Hub?
> 
> Hi Sayeed, this should be included in the `edge` containers (which are built nightly), and will be included with the next release (7.3.2?). For OpenShift, I'd recommend trying specifically `splunk/splunk:edge-redhat`.

Hi Mike. Thanks for the info. I'll test out the edge image. Is there an edge-redhat image for the the universal forwarder?

**Pull request merged**